### DISCUSSION TITLE
How to discard fields in WebApiAuditAction

### DISCUSSION THREAD

**Discussion by RajanModha** [thepirat000/Audit.NET#453] [Q&A]
I want to discard some fields like UserName, FormVariables etc. from Action of webApiAuditAction I'm able to set them to null but I want them to be discarded from my audit. 
Also I'm using Audit.WebApi along with Audit.EntityFramework so both of them are creating audit events and due to that I'm having 2 audit logs for 1 request I want to append it in single event such that I can get only one log

Waiting for your help :)

**Comment by thepirat000** [comment]
There is no easy way of composing different audit event types into one. 

I wonder why are you looking to have just one log per request? The usual way is to **correlate** the different events of a single request. 

For that, you could include the HttpContext's correlation ID (TraceIdentifier) on all the events via a Custom Action, so the events share a common unique ID per request. 

Actually, the WebApi audit event already includes the TraceIdentifier by default on the TraceId field

**Reply by thepirat000** [reply]
please check https://github.com/thepirat000/Audit.NET/issues/196#issuecomment-458283549

**Reply by RajanModha** [reply]
Thanks a lot for support. It really helped :)

**Reply by RajanModha** [reply]
Just a quick question though, the behavior of this package is async right? I mean to ask is auditing will affect my API performance?

**Reply by thepirat000** [reply]
The behavior sync/async depends on what are you auditing, for example if you call your DbContext.SaveChangesAsync(), it will execute asynchronously. If you call SaveChanges() instead, it will be synchronous. 
Regardless of the sync/async, of course the auditing will affect performance, since your app will be doing more work, so you should consider benchmarking the performance with and without audit enabled
